funding/commitment_type_negotiation.go

```diff
@@ -45,23 +46,62 @@ func negotiateCommitmentType(channelType *lnwire.ChannelType,
 		// user doesn't end up with an unexpected channel type via
 		// implicit negotiation.
 		if mustBeExplicit {
-			return false, 0, errUnsupportedExplicitNegotiation
+			return false, 0, false,
+				errUnsupportedExplicitNegotiation
 		}
 	}
 
-	return false, implicitNegotiateCommitmentType(local, remote), nil
+	implicitCommitType := implicitNegotiateCommitmentType(local, remote)
+	return false, implicitCommitType, false, nil
 }
 
 // explicitNegotiateCommitmentType attempts to explicitly negotiate for a
 // specific channel type. Since the channel type is comprised of a set of even
 // feature bits, we also make sure each feature is supported by both peers. An
-// error is returned if either peer does not support said channel type.
+// error is returned if either peer does not support said channel type. A bool
+// is returned indicating whether a zero-conf channel was negotiated.
 func explicitNegotiateCommitmentType(channelType lnwire.ChannelType,
-	local, remote *lnwire.FeatureVector) (lnwallet.CommitmentType, error) {
+	local, remote *lnwire.FeatureVector) (lnwallet.CommitmentType, bool,
+	error) {
 
 	channelFeatures := lnwire.RawFeatureVector(channelType)
 
 	switch {
+	// Lease script enforcement + anchors zero fee + static remote key +
+	// zero conf features only.
+	case channelFeatures.OnlyContains(
+		lnwire.ZeroConfRequired,
+		lnwire.ScriptEnforcedLeaseRequired,
+		lnwire.AnchorsZeroFeeHtlcTxRequired,
+		lnwire.StaticRemoteKeyRequired,
+	):
+		if !hasFeatures(
+			local, remote,
+			lnwire.ScidAliasOptional,
+			lnwire.ScriptEnforcedLeaseOptional,
+			lnwire.AnchorsZeroFeeHtlcTxOptional,
+			lnwire.StaticRemoteKeyOptional,
+		) {
+			return 0, false, errUnsupportedChannelType
+		}
+		return lnwallet.CommitmentTypeScriptEnforcedLease, true, nil
```

Comment:
Same comment here re the bool vs just starting to return the feature bit vector. 

I think we'll need to make that `unit8` -> `BigSize` conversion here in order to make it possible.

Reply:
starting to return feature bit vector?

Reply:
did it without needing any conversion